### ISSUE TITLE
Support flags and options for cloning

### ISSUE THREAD

**Issue by siketyan** [siketyan/ghr#36]
We need support for flags and options on cloning a repository. e.g. Cloning submodule recursively.

**Comment by foolishell** [comment]
### About `--bare` option
Original `git clone --bare <url>` command perfom like this.

```
$ git clone --bare git@github.com:siketyan/ghr.git
```
then, `./ghr.git` (not `./.ghr`) will be created. 

Is ghr going to be same with it?
It means we can create both: `/Users/USERNAME/github.com/siketyan/ghr.git` by bare option and `/Users/USERNAME/github.com/siketyan/ghr` by non-bare option.

**Comment by siketyan** [comment]
@foolishell 

Yes, ghr is going to do so. Bare repositories have no working tree, so opening and cd-ing the repo features will be disabled implicitly. Also we need to handle the different directory name on attaching the profile if --bare option is provided (this can be omitted in the first implementation).
